dockstore-common/src/main/java/io/dockstore/common/S3ClientHelper.java

```diff
@@ -67,15 +73,22 @@ public static S3Client createS3Client() {
      * @return
      */
     public static S3Client createS3Client(Region region) {
-        return initS3ClientBuilder().region(region).build();
+        return S3_CLIENT_MAP.computeIfAbsent(region, k -> initS3ClientBuilder().region(region).build());
     }
 
     /**
      * This should only be used by tests so that the test can provide a localstack endpoint override
      */
     public static S3Client createS3Client(String endpointOverride) throws URISyntaxException {
         LOG.info("Using endpoint override: {}", endpointOverride);
-        return initS3ClientBuilder().endpointOverride(new URI(endpointOverride)).build();
+        synchronized (S3_CLIENT_MAP) {
```

Comment:
This section appears to be caching and reusing the S3Clients by endpoint override.  Does it appreciably improve (or enable future improvements of) any aspect of the system?  If not, suggest removing, because it makes things significantly more complex.

Reply:
Hmmm, this method is used for testing. (according to the comment)
I guess the assumption was that we would have one s3client for aws in prod but in testing, we wanted isolated clients on different endpoints? Did the tests fail without this complexity?

Reply:
> Hmmm, this method is used for testing. (according to the comment)

Yeah, this is used for the integration tests (or local testing) so that the S3 client requests are directed at localstack using the endpoint override:

https://github.com/dockstore/dockstore/blob/f34564a16ad5af83963ad12c506b1da50f52957f/dockstore-integration-testing/src/test/resources/dockstore.yml#L56-L57